siliconcompiler/core.py

```diff
@@ -2952,57 +2953,76 @@ def summary(self, steplist=None, show_all_indices=False):
+            if generate_html or generate_pdf:
+                env = Environment(loader=FileSystemLoader(templ_dir))
+            if generate_html:
+                schema = self.schema.copy()
+                schema.prune()
+                pruned_cfg = schema.cfg
+                if 'history' in pruned_cfg:
+                    del pruned_cfg['history']
+                if 'library' in pruned_cfg:
+                    del pruned_cfg['library']
+
+                # Hardcode the encoding, since there's a Unicode character in a
+                # Bootstrap CSS file inlined in this template. Without this setting,
+                # this write may raise an encoding error on machines where the
+                # default encoding is not UTF-8.
+                with open(results_html, 'w', encoding='utf-8') as wf:
+                    wf.write(env.get_template('sc_report.j2').render(
+                        design = design,
+                        nodes = nodes,
+                        errors = errors,
+                        metrics = metrics,
+                        reports = reports,
+                        manifest = self.schema.cfg,
+                        pruned_cfg = pruned_cfg,
+                        metric_keys = metrics_to_show,
+                        img_data = img_data,
+                    ))
+
+            if generate_pdf:
+                # TODO: Add LaTeX generation logic or find a way to create similar PDFs with Sphinx/RST.
+                pass
 
             # Try to open the results and layout only if '-nodisplay' is not set.
-            if not self.get('option', 'nodisplay'):
-                try:
-                    webbrowser.get(results_page)
-                except webbrowser.Error:
-                    # Python 'webbrowser' module includes a limited number of popular defaults.
-                    # Depending on the platform, the user may have defined their own with $BROWSER.
-                    if 'BROWSER' in os.environ:
-                        subprocess.run([os.environ['BROWSER'], results_page])
+            # Prioritize PDF, followed by HTML.
+            self.logger.warning(results_pdf)
+            if (not self.get('option', 'nodisplay')) and ('DISPLAY' in os.environ):
+                if os.path.isfile(results_pdf):
+                    self.logger.warning('EXISTS')
+                    # Open results with whatever application is associated with PDFs on the local system.
+                    if sys.platform == 'win32':
+                        os.startfile(results_pdf)
+                    elif sys.platform == 'darwin':
+                        subprocess.Popen(['open', results_pdf])
                     else:
-                        self.logger.warning('Unable to open results page in web browser:\n' +
-                                            os.path.abspath(os.path.join(web_dir, "report.html")))
+                        subprocess.Popen(['xdg-open', results_pdf])
```

Comment:
why not use startfile for all? seems like python recommends that

Reply:
The call only exists on windows.